### ISSUE TITLE
Cannot load PSReadline module

### ISSUE THREAD

**Issue by dgriffen** [Microsoft/WhackWhackTerminal#10]
_From @AArnott on July 26, 2017 5:7_

When I opened the terminal tool window for the first time, I saw this error:
![image](https://user-images.githubusercontent.com/3548/28605299-b2475fc6-7185-11e7-8808-05dabb60e60a.png)

_Copied from original issue: dgriffen/WhackWhackTerminal#4_

**Comment by dgriffen** [comment]
I have not been able to reproduce this issue, is PSReadLine an external module? If so it could be an issue where the powershell we launch uses some different profile settings. I have had success loading posh-git so I know now all modules are affected.

**Comment by dgriffen** [comment]
_From @Tyriar on August 3, 2017 5:25_

This happens when you run the 32-bit version of powershell.exe on a 64-bit machine. If Visual Studio is a 32-bit executable you want to launch `%windir%\Sysnative\WindowsPowerShell\\v1.0\\powershell.exe`. If Visual Studio is a 64-bit executable, you want to launch `%windir%\System32\WindowsPowerShell\\v1.0\\powershell.exe`.

**Comment by dgriffen** [comment]
_From @thomaslevesque on August 3, 2017 23:27_

I got the same error. Also, it doesn't seem to execute my PS profile, and if I try to execute it manually, I get this error:

```
PS D:\Git\FakeItEasy> . $PROFILE
. : File C:\Users\Tom\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1 cannot be
loaded because running scripts is disabled on this system. For more information, see
about_Execution_Policies at http://go.microsoft.com/fwlink/?LinkID=135170.
```

But my execution policy is "Unrestricted"...

EDIT: @Tyriar's explanation is correct. My execution policy for 64-bit PS was Unrestricted, but for 32-bit PS it was still Restricted, because I never use it... To fix, run Powershell (x86) as admin and run `Set-ExecutionPolicy Unrestricted`

**Comment by dgriffen** [comment]
@thomaslevesque you could also add the `-Scope CurrentUser` flag to run the `Set-ExecutionPolicy` command in the integrated terminal. That way you don't need admin privileges. I would love to be able to avoid all these issues entirely but I'm limited by the fact that the node runtime VS gives me is 32bit.

Edit: just read @Tyriar's response in its entirety, guess I could set the program to execute the 64bit powershell by default.

**Comment by dgriffen** [comment]
_From @Tyriar on August 4, 2017 1:32_

@dgriffen that's the solution I landed on. Look here for how to set the right default https://github.com/Microsoft/vscode/blob/a7c10fea150543ac0d159c08751f24899d8b8131/src/vs/workbench/parts/terminal/electron-browser/terminal.ts#L17

We're actually experiencing the opposite problem right now where we've been advising to use Sysnative for over a year and now there's a 64-bit build which won't work at all with a Sysnative path ([just wrote a blog post about it](http://www.growingwiththeweb.com/2017/08/windows-wow64.html)). So now we're just forcing System32 when Wow64 is active https://github.com/Microsoft/vscode/blob/a7c10fea150543ac0d159c08751f24899d8b8131/src/vs/workbench/parts/terminal/electron-browser/terminalConfigHelper.ts#L160

**Comment by dgriffen** [comment]
Closed by #4
